src/main/java/com/o19s/es/explore/ExplorerScorer.java

```diff
@@ -0,0 +1,133 @@
+/*
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.o19s.es.explore;
+
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.index.PostingsEnum;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.index.Terms;
+import org.apache.lucene.index.TermsEnum;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.search.Scorer;
+import org.apache.lucene.search.Weight;
+import org.apache.lucene.search.similarities.ClassicSimilarity;
+import org.elasticsearch.common.logging.Loggers;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ExplorerScorer extends Scorer {
+    private Scorer subscorer;
+    private String field, type;
+    private LeafReaderContext context;
+
+    protected ExplorerScorer(Weight weight, LeafReaderContext context, String field, String type, Scorer subscorer) {
+        super(weight);
+        this.context = context;
+        this.field = field;
+        this.type = type;
+        this.subscorer = subscorer;
+    }
+
+    @Override
+    public float score() throws IOException {
+        // TODO: Cleanup DRY violation, similar code in the Explorer fetch phase
+        Set<Term> terms = new HashSet<Term>();
+        weight.extractTerms(terms);
+
+        // No stats if no terms
+        if(terms.size() == 0) return 0.0f;
+
+        Terms termVector = context.reader().getTermVector(docID(), field);
```

Comment:
(forgot to add this comment in my previous review, sorry)

I can see 2 drawbacks of using the termVector for these kind of stats:
- it's rather slow
- it's not always available

In our context we do not index them and we might have to hack our mappings and reindex to take benefit of your new query (@ebernhardson might have some opinion on this?).
We could generate termVectors on the fly but this would be even more slow and complex.

I'm not strongly against using termVectors for a first iteration (to decide if these features are useful) but I'm not too keen running this kind of query in production, it'll fetch a rather large portion of data from disk on a lot of docs.

Also I don't think it's very complex to extract the stats we want without using termVectors.
First thing is to do is to distinguish the scope of stat source:
- Collection scoped: the score is the the same for all docs a constant scorer might be sufficient
  - `raw_df`: `TermStatistics#docFreq()`
  - `raw_ttf`: `TermStatistics#sumTotalFreq()`
- Doc scoped: different on every doc, a custom scorer like this one is needed
  - `raw_tf`: `PostingEnum#freq()`: 
    - tip: here you use subscorer and I think you can easily fetch subterms freq by iterating on `subscorer.getChildren()`:
```java
        for (ChildScorer child : subscorer.getChildren()) {
            if (child.child.docID() == docID()) {
                tf_stats.add(child.child.freq());
            }
        }
```
`Scorer#getChildren()` highly depends on the nested query we pass, but for simple `match_query` it should work (not tested).